### PR TITLE
build(ci): optimize Cargo `release` for program

### PR DESCRIPTION
**Pull request by Rudxain**
https://doc.rust-lang.org/cargo/reference/profiles.html

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -46,8 +46,10 @@ win32console = "^0.1.5"
 
 [profile.release]
 opt-level = "s"
+codegen-units = 1
 lto = true
-strip = "symbols"
+strip = true
```
Comment (Rudxain) [inline Cargo.toml:51]: `strip` `true` is more consistent with `lto`

**Comment by Rudxain** [comment]
Why is Cargo CI? 😕

**Comment by AnonymousWP** [comment]
Because the `Cargo.toml` is used when building the assets, right? And assets are being build when a tag is pushed for a new version.🙂 Let me know if that way of thinking is incorrect.

File: Cargo.toml [hunk 2]
```diff
@@ -46,8 +46,10 @@ win32console = "^0.1.5"
 
 [profile.release]
 opt-level = "s"
+codegen-units = 1
```
Comment (adhirajsinghchauhan) [inline Cargo.toml:49]: This is sure to lead to much slower code. IMO remove this line, it doesn't really help much with executable sizes, so let it be at the default of `16`.

File: Cargo.toml (hunk 1 above)
Comment (adhirajsinghchauhan) [reply Cargo.toml:51]: Yes, `"symbols"` and `true` are the same thing, but it could change in future rust releases. I'm fine either way.

**Comment by Rudxain** [comment]
> `Cargo.toml` is used when building the assets

Correct. But Cargo is also used outside of CI, such as local environments.

I suppose we can keep it as "CI", as the [Wikipedia definition](https://en.wikipedia.org/wiki/Continuous_integration) seems to include build systems 🤔

**Review by adhirajsinghchauhan — CHANGES_REQUESTED** [review]

**Comment by adhirajsinghchauhan** [comment]
I'm interested to see the size differences between panic `unwind` and `abort`. Could you post comparisons by building the latest release with both options?

File: Cargo.toml (hunk 2 above)
Comment (Rudxain) [reply Cargo.toml:49]: I could only find this rust-lang/rust#47745 , but I'm not sure how to interpret all the linked issues 😕

Should we set it to 4 instead?

Comment (adhirajsinghchauhan) [reply Cargo.toml:49]: I believe that's specific to `lto="thin"`, which we don't use. We don't have any benchmarks for code performance, so the only test we can do is build with and without defaults codegen values, and compare file sizes.

**Comment by Rudxain** [comment]
`cargo build --release; stat -c%s target/release/uad-ng` on 3 branches:
- `main` 16261816
- `build/opt-release` 14346792
- `default-codegen-units` (local) 14748040

**Review by adhirajsinghchauhan — APPROVED** [review]

**Comment by Rudxain** [comment]
[This](https://github.com/rust-lang/cargo/issues/11298) is also worth referencing

**Comment by AnonymousWP** [comment]
image

Perhaps we need to do this.

**Comment by Rudxain** [comment]
@AnonymousWP wait! I meant "reference" as the thing I just did, which is "to mention it with a link". It's just to provide more info about the subject, as optimization is hard to get right (sometimes)

**Pull request merged**